doc/performance_blog_post/performance_blog_post.md

```diff
@@ -0,0 +1,109 @@
+
+The first test run generated the following results:
+
+| Test Case                          | Time in s |
+|------------------------------------|-----------|
+| testLoadManyParquetRows            | 23        |
+| testLoadLargeParquetRows           | 13.02     |
+| testLoadManyParquetColumns         | 12.9      |
+| testLoadManyParquetRowsFromOneFile | 74.64     |
+| testManySmallJsonFiles             | 685.87    |
+| testLoadFewBigSalesParquetFiles    | 662.28    |
+| testLoadSalesParquetFiles          | 219.21    |
+
+The first 4 cases (`testLoadManyParquetRows`, `testLoadLargeParquetRows`, `testLoadManyParquetColumns` `testLoadManyParquetRowsFromOneFile`) all load the same amount of data. They just differ on how the data is distributed over rows, files and columns.
+
+The same applies for the two trailing tests: While `testLoadFewBigSalesParquetFiles` distributes the rows over 200 files, `testLoadSalesParquetFiles` only over 8 files.
+
+![Performance test results of parquet file loading 1](slowParquetTestResults1.png)
+![Performance test results of parquet file loading 2](slowParquetTestResults2.png)
+
+In the charts it becomes clear that the Virtual Schema at that point has an issue with loading data from few big files. When loading from more files, it loads the same amount of data a lot quicker.
+
+Starting with this analysis, we discussed with our product management and decided to invest into performance improvements for loading big parquet files. More precisely we set the goal to make improvements so that the `testLoadFewBigSalesParquetFiles` also runs in less than 250s.
+
+## Improving the Performance
+
+Next we analyzed the source code for possible reasons for the slow reading of big parquet files. There we identified the following issues:
+
+* Unequal distribution
+* Too few parallelization for few parquet files
+
+The query processing of the Files Virtual Schemas consists of two phases: A planning phase and an execution phase. In the planning phase the adapter analyzes the query and starts multiple workers (UDFs) that then run the execution phase. For technical reasons the workers can not communicate at the moment. So the Virtual Schema parallelizes the import job over multiple workers.
+
+### Distributing the File
+
+Until now this parallelization was realized using hash binning. So each worker received a range and then loaded files that names hash values fell into that range. That approach is great for distributing lots of files, since it does not require to list all files in advance. Doing so could cause memory overflow when distributing for example 1,000,000 files with long names.
+
+However, for only a few files, this works rather bad:
+Imagine distributing 4 files over 4 workers. A good distribution is obvious here. Each worker receives on file. The workers then load the files in parallel. With hash binning it's however quite likely, that one worker receives two files and another none. This will increase the time for loading.
```

Comment:
```suggestion
Imagine distributing 4 files over 4 workers. A good distribution is obvious here. Each worker receives one file. The workers then load each file in parallel. With hash binning it's however quite likely, that one worker receives two files and another receives none. This will increase the time for loading.
```